src/linpred.jl

```diff
@@ -336,37 +422,77 @@ function modelframe(f::FormulaTerm, data, contrasts::AbstractDict, ::Type{M}) wh
     f, modelcols(f, data)
 end
 
-modelmatrix(obj::LinPredModel) = obj.pp.X
+modelframe(obj::LinPredModel) = obj.fr
+
+modelmatrix(obj::LinPredModel; weighted::Bool=isweighted(obj)) = modelmatrix(obj.pp; weighted=weighted)
+
+function modelmatrix(pp::LinPred; weighted::Bool=isweighted(pp))
+    Z = if weighted
+        Diagonal(sqrt.(pp.wts))*pp.X
+    else
+        pp.X
+    end
+    return Z
+end
+
+leverage(x::LinPredModel) = leverage(x.pp)
+
+function leverage(pp::DensePredChol{T, C, W}) where {T, C<:CholeskyPivoted, W}
+    X = modelmatrix(pp; weighted=isweighted(pp))
+    _, k = size(X)
+    ch = pp.chol
+    rnk = rank(ch)
+    p = ch.p
+    idx = invperm(p)[1:rnk]
+    sum(x -> x^2, view(X, :, 1:rnk)/ch.U[1:rnk, idx], dims=2)
+end
+
+function leverage(pp::DensePredChol{T, C, W}) where {T, C<:Cholesky, W}
+    X = modelmatrix(pp; weighted=isweighted(pp))
+    sum(x -> x^2, X/pp.chol.U, dims=2)
+end
+
+function leverage(pp::DensePredQR{T, C, W}) where {T, C<:QRPivoted, W}
+    X = modelmatrix(pp; weighted=isweighted(pp))
+    _, k = size(X)
+    ch = pp.qr
+    rnk = length(ch.p)
+    p = ch.p
+    idx = invperm(p)[1:rnk]
+    sum(x -> x^2, view(X, :, 1:rnk)/ch.R[1:rnk, idx], dims=2)
+end
+
+function leverage(pp::DensePredQR{T, C, W}) where {T, C<:Cholesky, W}
+    X = modelmatrix(pp; weighted=isweighted(pp))
+    sum(x -> x^2, X/pp.qr.R, dims=2)
+end
+
 response(obj::LinPredModel) = obj.rr.y
 
 fitted(m::LinPredModel) = m.rr.mu
 predict(mm::LinPredModel) = fitted(mm)
-residuals(obj::LinPredModel) = residuals(obj.rr)
 
 function StatsModels.formula(obj::LinPredModel)
     obj.formula === nothing && throw(ArgumentError("model was fitted without a formula"))
     return obj.formula
 end
 
-"""
-    nobs(obj::LinearModel)
-    nobs(obj::GLM)
+residuals(obj::LinPredModel; weighted::Bool=false) = residuals(obj.rr; weighted=weighted)
 
-For linear and generalized linear models, returns the number of rows, or,
-when prior weights are specified, the sum of weights.
-"""
```

Comment:
Looks like you have removed this docstring, which explains why CI fails when building docs.